xivo/tenant_helpers.py

```diff
@@ -148,6 +135,27 @@ def tenant_uuid(self) -> str | None:
     def user_uuid(self) -> str | None:
         return self._token_dict['metadata'].get('uuid')
 
+    @property
+    def _token_dict(self) -> dict[str, Any]:
+        if self.__token_dict is None:
+            try:
+                self.__token_dict = self._auth.token.get(self.uuid)
+            except requests.HTTPError:
+                raise InvalidToken(self.uuid)
+            except requests.RequestException as e:
+                raise AuthServerUnreachable(self._auth.host, self._auth.port, e)
+
+        return self.__token_dict
+
+    def has_tenant_access(self, tenant_uuid: str | None) -> bool:
+        if not tenant_uuid:
+            return False
+
+        try:
+            return self._auth.token.is_valid(self.uuid, tenant=tenant_uuid)
```

Comment:
A slight optimization is to check if we have the tenant in `__token_dict` before starting the request to wazo-auth.

Reply:
Done